components/features/currency/currency-converter/hooks/use-currency-converter.tsx

```diff
@@ -0,0 +1,70 @@
+import { useEffect, useState } from "react";
+import { Money } from "utils/Money/Money";
+
+import { TCurrencyConverter } from "components/features/currency/currency-converter/currency-converter.types";
+import { TUseCurrencyConverter } from "components/features/currency/currency-converter/hooks/use-currency-converter.types";
+
+export function useCurrencyConverter({ budgets }: TUseCurrencyConverter.Props) {
+  const [usdValue, setUsdValue] = useState<string>("");
+  const [currencyAmount, setCurrencyAmount] = useState<string>("");
```

Comment:
Explicit types here muddy the code a little

```suggestion
  const [usdValue, setUsdValue] = useState("");
  const [currencyAmount, setCurrencyAmount] = useState("");
```